langchain/vectorstores/marqo.py

```diff
@@ -0,0 +1,413 @@
+        verbose: bool = True,
+        **kwargs: Any,
+    ) -> Marqo:
+        """Return Marqo initialized from texts. Note that Marqo does not need embeddings, we retain the
+        parameter to adhere to the Liskov substitution principle.
+
+        This is a quick way to get started with marqo - simply provide your texts and metadatas and this
+        will create an instance of the data store and index the provided data.
+
+        To know the ids of your documents with this approach you will need to include them in under the key "_id"
+        in your metadatas for each text
+
+        Example:
+        .. code-block:: python
+
+                from langchain.vectorstores import Marqo
+
+                datastore = Marqo(texts=['text'], index_name='my-first-index', url='http://localhost:8882')
+
+        Args:
+            texts (List[str]): A list of texts to index into marqo upon creation.
+            embedding (Any, optional): Embeddings (not required). Defaults to None.
+            index_name (str, optional): The name of the index to use, if none is provided then one will be created with a UUID. Defaults to None.
+            url (str, optional): The URL for Marqo. Defaults to "http://localhost:8882".
+            api_key (str, optional): The API key for Marqo. Defaults to "".
+            metadatas (Optional[List[dict]], optional): A list of metadatas, to accompany the texts. Defaults to None.
+            marqo_device (str, optional): The device for the marqo to use on the server, this is only used when a new index is being created. Defaults to "cpu".
+            add_documents_settings (Optional[Dict[str, Any]], optional): Settings for adding documents, see https://docs.marqo.ai/0.0.16/API-Reference/documents/#query-parameters. Defaults to {}.
+            index_settings (Optional[Dict[str, Any]], optional): Index settings if the index doesn't exist, see https://docs.marqo.ai/0.0.16/API-Reference/indexes/#index-defaults-object. Defaults to {}.
+
+        Returns:
+            Marqo: An instance of the Marqo vector store
+        """
+        try:
+            import marqo
+        except ImportError:
+            raise ValueError(
+                "Could not import marqo python package. "
+                "Please install it with `pip install marqo`."
+            )
+
+        if not index_name:
+            index_name = str(uuid.uuid4())
+
+        client = marqo.Client(url=url, api_key=api_key, indexing_device=marqo_device)
+
+        try:
+            client.create_index(index_name, settings_dict=index_settings)
+            if verbose:
+                print(f"Created {index_name} successfully.")
+        except:
+            if verbose:
+                print(f"Index {index_name} exists.")
+
+        instance: Marqo = cls(
+            client,
+            index_name,
+            searchable_attributes=searchable_attributes,
+            add_documents_settings=add_documents_settings,
```

Comment:
should kwargs be passed in as well?

Reply:
I don't think so because that is all the arguments required, kwargs is not passed through to the constructors on any of the other vectorstore implementations either